editor/src/messages/portfolio/document/document_message_handler.rs

```diff
@@ -419,7 +420,13 @@ impl MessageHandler<DocumentMessage, DocumentInputs<'_>> for DocumentMessageHand
 
 					let id = NodeId(generate_uuid());
 					let insert_index = -1;
-					responses.add(GraphOperationMessage::NewCustomLayer { id, nodes, parent, insert_index });
+					responses.add(GraphOperationMessage::NewCustomLayer {
+						id,
+						nodes,
+						parent,
+						insert_index,
+						alias: String::from(""),
```

Comment:
You can use `String::new()` instead, which does not allocate memory

Reply:
Sure

Reply:
done

Reply:
@TrueDoctor I'm pretty sure `String::from("")` won't allocate - it looks like it just transmutates the string into a byte slice and then converts it to a Vec which will not allocate if the size is 0.

Reply:
@0HyperCube  In debug mode it allocates, when compiling with optimizations the same assembly is produced so I guess it is just good practice to use `String::new()` to guarantee it won't allocate + speed up debug builds but will most likely not make difference in production: https://godbolt.org/z/9r5o7jrs4